imgui_draw.cpp

```diff
@@ -3583,18 +3584,276 @@ void ImFont::RenderText(ImDrawList* draw_list, float size, const ImVec2& pos, Im
     if (s == text_end)
         return;
 
+    // draw_list_text 
+    ImDrawList* draw_list_text = nullptr;
+
+
+    ImDrawList _draw_list_text(ImGui::GetDrawListSharedData());
```

Comment:
result of `ImGui::GetDrawListSharedData()` could be put into the `if (customization)` scope

Reply:
The ImGui::GetDrawListSharedData() is moved to if (customization) scope. The cost is to new and delete a ImDrawList when drawing the text with customization.

Reply:
That mentioned cost + the fact it means draw list buffers will realloc+copy every frame and not be amortized is very largely in the range of unacceptably slow in Dear ImGui world. It makes me worry about the veracity of your other statement about performances.

Reply:
If the alloc/free is expensive we can revert to the old logic. The old logic creates a draw_list_text on the **stack** this may save some time for us. 

For the copy logic adding text draw list to the main draw_lists is shown as below. The text draw list has to be created anyway, so the cost is the same. The additional cost is the copy logic. 

1. The VtxBuffer is copied with memcpy should be fast. 
2. The IdxBuffer is copied with a for loop, may cost a few CPU clocks, but it cannot be avoided. If you have good suggestion, please advise. 

```    
    if (customization)
    {    
       ...
       // Append draw_list_text to the main draw_lists
        {
            unsigned int old_inx_buffer_sz = draw_list->IdxBuffer.Size;
            unsigned int old_vtx_buffer_sz = draw_list->VtxBuffer.Size;
            draw_list->PrimReserve(draw_list_text->IdxBuffer.Size, draw_list_text->VtxBuffer.Size);
            // draw_list_text->VtxBuffer buffer can be memcopied, it contains no relative data
            memcpy(draw_list->_VtxWritePtr, draw_list_text->VtxBuffer.Data, draw_list_text->VtxBuffer.Size * sizeof(ImDrawVert));
            // draw_list_text->IdxBuffer buffer cann't because the index value in the IdxBuffer points to draw_list_text->VtxBuffer. It need aligned to draw_list->VtxBuffer.
            for (int i = 0; i < draw_list_text->IdxBuffer.Size; i++)
            {
                draw_list->IdxBuffer[old_inx_buffer_sz + i] = (ImDrawIdx)(old_vtx_buffer_sz + draw_list_text->IdxBuffer[i]);
            }
            draw_list->_VtxWritePtr += draw_list_text->VtxBuffer.Size;
            draw_list->_IdxWritePtr += draw_list_text->IdxBuffer.Size;
            draw_list->_VtxCurrentIdx += vtx_current_idx;
        }

     ...
```

Reply:
> That mentioned cost + the fact it means draw list buffers will realloc+copy every frame and not be amortized is very largely in the range of unacceptably slow in Dear ImGui world. It makes me worry about the veracity of your other statement about performances.

Is there a performance benchmark for text, it would be a good idea to test it and compare the performance results.